src/envoy/mixer/grpc_transport.cc

```diff
@@ -72,6 +82,18 @@ bool GrpcTransport::IsMixerServerConfigured(Upstream::ClusterManager& cm) {
   return cm.get(kMixerServerClusterName) != nullptr;
 }
 
+void GrpcTransport::onPreRequestCustomizeHeaders(Http::HeaderMap& headers) {
+  if (!headers_) return;
+
+  for (const auto& key : kHttpTraceHeaders) {
+    const HeaderEntry* entry = headers_->get(key);
```

Comment:
AFAIK get doesn't work for inline headers (e.g. every header other than "x-ot-span-context"), this might be an issue for envoy itself.
https://github.com/lyft/envoy/blob/master/source/common/http/header_map_impl.cc#L338

Do you have a test to confirm the behavior of this PR?